### PR TITLE
Update CreateComponentTool.cs

### PR DESCRIPTION
**Pull request by hjqcan**
1.Allow for the creation of components with parameters 
2.Add event when component creation is complete
3.buf fix for BasicMetadata.Register() never use

### CODE REVIEW

File: WpfDesign.Designer/Project/Services/CreateComponentTool.cs [hunk 1]
```diff
@@ -220,9 +228,9 @@ protected virtual DesignItem[] CreateItems(DesignContext context)
 		/// <summary>
 		/// Is called to create the item used by the CreateComponentTool.
 		/// </summary>
-		public static DesignItem CreateItem(DesignContext context, Type type)
+		public static DesignItem CreateItem(DesignContext context, Type type, object[] arguments)
```
Comment (jogibear9988) [inline WpfDesign.Designer/Project/Services/CreateComponentTool.cs:231]: same here, add additional static method

File: WpfDesign.Designer/Project/Services/CreateComponentTool.cs [hunk 2]
```diff
@@ -26,26 +26,31 @@
...
-		/// Creates a new CreateComponentTool instance.
-		/// </summary>
-		public CreateComponentTool(Type componentType)
+
+		public event EventHandler<DesignItem> CreateComponentCompleted;
+
+        /// <summary>
+        /// Creates a new CreateComponentTool instance.
+        /// </summary>
+        public CreateComponentTool(Type componentType, object[] arguments)
```
Comment (jogibear9988) [inline WpfDesign.Designer/Project/Services/CreateComponentTool.cs:47]: with this it is a breaking change.
Please create a additional constructor without the perameter

Comment (hjqcan) [reply WpfDesign.Designer/Project/Services/CreateComponentTool.cs]: sorry about that,done ser.

File: WpfDesign.Designer/Project/Services/CreateComponentTool.cs (hunk 1 above)
Comment (hjqcan) [reply WpfDesign.Designer/Project/Services/CreateComponentTool.cs]: done ser.changed to  public static DesignItem CreateItem(DesignContext context, Type type, object[] arguments=null)

File: WpfDesign.Designer/Project/Services/CreateComponentTool.cs [hunk 3]
```diff
@@ -220,9 +239,9 @@ protected virtual DesignItem[] CreateItems(DesignContext context)
 		/// <summary>
 		/// Is called to create the item used by the CreateComponentTool.
 		/// </summary>
-		public static DesignItem CreateItem(DesignContext context, Type type)
+		public static DesignItem CreateItem(DesignContext context, Type type, object[] arguments = null)
```
Comment (jogibear9988) [inline WpfDesign.Designer/Project/Services/CreateComponentTool.cs:242]: no, create an overload. even with a default value, it is a breaking change wich needs recompilation. 

the first method without arguments can then call the new one with

File: WpfDesign.Designer/Project/Services/CreateComponentTool.cs [hunk 4]
```diff
@@ -235,15 +254,15 @@ public static DesignItem CreateItem(DesignContext context, Type type)
...
-		public static bool AddItemWithCustomSizePosition(DesignItem container, Type createdItem, Size size, Point position)
+		public static bool AddItemWithCustomSizePosition(DesignItem container, Type createdItem, object[] arguments, Size size, Point position)
 		{
-			CreateComponentTool cct = new CreateComponentTool(createdItem);
+			CreateComponentTool cct = new CreateComponentTool(createdItem, arguments);
 			return AddItemsWithCustomSize(container, new[] { cct.CreateItem(container.Context) }, new[] { new Rect(position, size) });
 		}
 		
-		public static bool AddItemWithDefaultSize(DesignItem container, Type createdItem, Size size)
+		public static bool AddItemWithDefaultSize(DesignItem container, Type createdItem, object[] arguments, Size size)
```
Comment (jogibear9988) [inline WpfDesign.Designer/Project/Services/CreateComponentTool.cs:263]: also here, breaking change, create an overload

File: WpfDesign.Designer/Project/Services/CreateComponentTool.cs [hunk 5]
```diff
@@ -235,15 +254,15 @@ public static DesignItem CreateItem(DesignContext context, Type type)
 		protected virtual void SetPropertiesForDrawnItem(DesignItem designItem)
 		{ }
 
-		public static bool AddItemWithCustomSizePosition(DesignItem container, Type createdItem, Size size, Point position)
+		public static bool AddItemWithCustomSizePosition(DesignItem container, Type createdItem, object[] arguments, Size size, Point position)
```
Comment (jogibear9988) [inline WpfDesign.Designer/Project/Services/CreateComponentTool.cs:257]: same here, breaking change, create an overload

File: WpfDesign.Designer/Project/Services/CreateComponentTool.cs (hunk 2 above)
Comment (jogibear9988) [reply WpfDesign.Designer/Project/Services/CreateComponentTool.cs]: the additional constructor can call the base, so we have no code duplicated

Comment (hjqcan) [reply WpfDesign.Designer/Project/Services/CreateComponentTool.cs]: Thank you for your thorough review.Had submited a new one

**Comment by jogibear9988** [comment]
thanks

**Pull request merged**